apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -37,16 +42,29 @@ class PageTemplateModal extends Component {
 		if ( this.state.isOpen ) {
 			trackView( this.props.segment.id, this.props.vertical.id );
 		}
+
+		// Populate blocks state field, parsing the raw content for each template.
+		const blocks = {};
+		for ( const slug in this.props.templates ) {
+			const template = this.props.templates[ slug ];
+			blocks[ slug ] = template.content ? parseBlocks( template.content ) : [];
+		}
+
+		// eslint-disable-next-line react/no-did-mount-set-state
+		this.setState( { blocks } );
+
 		// eslint-disable-next-line no-console
 		console.timeEnd( 'PageTemplateModal' );
 	}
 
-	setTemplate = ( slug, title, previewBlocks ) => {
+	setTemplate = ( slug, title ) => {
 		this.setState( { isOpen: false } );
 		trackSelection( this.props.segment.id, this.props.vertical.id, slug );
 
 		this.props.saveTemplateChoice( slug );
 
+		const previewBlocks = this.state.blocks[ slug ];
```

Comment:
Can this ever be `undefined`? If so we should have a fallback.

Reply:
Kinda impossible, however, let's add a method which can return a fallback is it's needed.